### PR TITLE
Replace EnvironmentError and IOError by OSError

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Refactoring

### Purpose
Since python 3.3, EnvironmentError and IOError were merged into OSError.
refs: https://docs.python.org/3/library/exceptions.html#OSError

### CODE REVIEW

File: sphinx/ext/graphviz.py [hunk 1]
```diff
@@ -261,7 +261,7 @@ def render_dot(self, code, options, format, prefix='graphviz'):
         # Graphviz may close standard input when an error occurs,
         # resulting in a broken pipe on communicate()
         stdout, stderr = p.communicate(code.encode('utf-8'))
-    except (OSError, IOError) as err:
+    except OSError as err:
```
Comment (jdufresne) [inline sphinx/ext/graphviz.py]: Given the `errno` check below, could this catch `BrokenPipeError` instead?

https://docs.python.org/3/library/exceptions.html#BrokenPipeError

Comment (tk0miya) [reply sphinx/ext/graphviz.py]: It seems better. but it only responds to `EPIPE`. So we still have to catch `OSError`, right?

Comment (jdufresne) [reply sphinx/ext/graphviz.py]: Yeah, that is right. So `OSError` is still necessary to handle the `EINVAL` case, so there isn't much point to changing it. Do you know what triggers an `EINVAL`? I tried to find more details, but didn't find anything specific.

Comment (tk0miya) [reply sphinx/ext/graphviz.py]: It comes from 9350b3b46b. The commit message says:

>This happen on Python2.7 and Graphviz version 2.12 (Mon Dec  4 22:04:37 UTC 2006)
on Windows when a not recognized renderer type is asked.

I don't know latest python and windows causes the `EINVAL` error. We have to confirm that before removing...

Comment (tk0miya) [reply sphinx/ext/graphviz.py]: The `EINVAL` has not been thrown at present. That was fixed at https://bugs.python.org/issue19612
https://github.com/python/cpython/blob/master/Lib/subprocess.py#L880

@gjo Thank you for advice!

Comment (tk0miya) [reply sphinx/ext/graphviz.py]: fixed.

**Review by jdufresne — APPROVED** [review]

**Review by shimizukawa — APPROVED** [review]
LGTM!

**Comment by tk0miya** [comment]
Merged. Thank you for reviewing!

**Pull request merged**